src/core/qgsabstractdatabaseproviderconnection.cpp

```diff
@@ -442,15 +442,65 @@ QStringList QgsAbstractDatabaseProviderConnection::QueryResult::columns() const
 
 QList<QList<QVariant> > QgsAbstractDatabaseProviderConnection::QueryResult::rows() const
 {
+  // mRowCount might be -1 (unknown)
+  while ( mResultIterator && ( mRowCount < 0 || mRows.count() < mRowCount ) )
+  {
+    const QVariantList row { mResultIterator->nextRow() };
+    if ( row.isEmpty() )
```

Comment:
@elpaso , playing with MSSQL, since this PR is merged, I have an infinite loop here when I expand MSSQL layers in the browser panel.